src/DefaultBuilder/src/WebApplication.cs

```diff
@@ -0,0 +1,243 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Hosting.Server;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+// REVIEW: Or just "Microsoft.AspNetCore" like WebHost?
+namespace Microsoft.AspNetCore.Builder
+{
+    /// <summary>
+    /// The web application used to configure the http pipeline, and routes.
+    /// </summary>
+    public sealed class WebApplication : IHost, IDisposable, IApplicationBuilder, IEndpointRouteBuilder, IAsyncDisposable
```

Comment:
And we've come full circle:
https://github.com/aspnet/AspNetKatana/blob/dev/src/Microsoft.Owin.Hosting/WebApp.cs

Reply:
This is better though